plugins/janus_audiobridge.c

```diff
@@ -1919,8 +1924,25 @@ void janus_audiobridge_incoming_rtp(janus_plugin_session *handle, int video, cha
 		pkt->ssrc = 0;
 		pkt->timestamp = ntohl(rtp->timestamp);
 		pkt->seq_number = ntohs(rtp->seq_number);
+		/* Any audio level extension to quickly check if this is silence? */
+		pkt->silence = FALSE;
+		if(participant->extmap_id > 0) {
+			int level = 0;
+			if(janus_rtp_header_extension_parse_audio_level(buf, len, participant->extmap_id, &level) == 0) {
+				pkt->silence = (level == 127);
```

Comment:
I'm personally not familiar enough with the way this level is formatted: I still haven't figured out how to convert that dBov value to a generic and linear volume identifier, for instance, something that may actually be very hard if it is indeed dependant on the saturation of codecs themselves. Working on the dBov value might be quicker, if one knows how to interpret them.

Not sure on what we should do with thresholds. I guess we could make that a property of an audioroom when you create it, so that you can decide per-room whether to be more aggressive or not. Another alternative is to make it a per-participant property, as different users may have different captures (some louder, some not). Of course nothing prevents us to have them both: if a participant doesn't have a custom value, the global room one is used; if the room doesn't have it either, 127 is used. Does this make sense per your use case?